### ISSUE TITLE
Optimized continuous asynchronous parallel execution of playbooks across a subset of inventory hosts

### ISSUE THREAD

**Issue by beglowac** [ansible/ansible#46649]
##### SUMMARY
This is a request for an ansible feature to support execution of a playbook against a set of hosts asynchronously and in parallel, such that, when executed against an inventory a subset of x number hosts will run the playbook simultaneously and as fast as possible (strategy = free). Should one host finish the playbook run, ansible will then automatically select another host from the inventory and run the playbook on that host. In this way, there will always be a subset of x number of hosts running the playbook at any given time, excepting when there are less than x number of hosts remaining to be processed.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
Optimized continuous asynchronous parallel execution of playbooks across a subset of inventory hosts

##### ADDITIONAL INFORMATION
This can be used, for example, to gracefully offline and patch hosts that are actively in use for business purposes. For example, in a hypothetical scenario, some hosts can take as long as 8 hours to become available to be made offline for patch processing whereas others might be available in as little as 5 minutes. In an environment in which only a small subset of hosts may be made offline at a time to avoid negatively impacting overall host resource availability, using existing Ansible art requires that the entire subset of hosts be made offline and patched prior to moving on to make offline and patch the next subset of the the same number of hosts. So each set in this hypothetical scenario can be forced to take as long as 8 hours to patch because of the delay introduced by patching one host. 

This feature would remove that block and allow each host that has been made offline and successfully patched to be replaced with a new inventory host that will be similarly setup to be made offline and patched, thus keeping the subset count of hosts concurrently being made offline and patched at the same selected subset size. 

One existing solution to solve this problem is external to ansible:

ansible-playbook <playbook name..yml --list-hosts 2>/dev/null | sed -r -n -e '/<hostmatchingstring>/,${p}' | xargs -P5 -n1 ansible-playbook <playbook name>.yml

It would be much more optimal if this functionality were provided as a native ansible feature, perhaps by extending some combination of the existing serial/strategy and/or fork functionality.

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/inventory](https://github.com/ansible/ansible/blob/devel/lib/ansible/inventory)
* [lib/ansible/modules/system/setup.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/setup.py)
* [lib/ansible/playbook/become.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/become.py)
* [lib/ansible/playbook/block.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/block.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bcoca** [comment]
sounds like you want https://docs.ansible.com/ansible/latest/plugins/strategy/host_pinned.html

**Comment by ansibot** [comment]
cc @davidobrien1985
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/inventory](https://github.com/ansible/ansible/blob/devel/lib/ansible/inventory)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/plugins/inventory/openstack.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/inventory/openstack.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/plugins/inventory/openstack.py -> https://galaxy.ansible.com/openstack/cloud

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md

**Comment by mkrizek** [comment]
!needs_collection_redirect

**Comment by mkrizek** [comment]
> sounds like you want https://docs.ansible.com/ansible/latest/plugins/strategy/host_pinned.html

needs_info

**Comment by ansibot** [comment]
@beglowac This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@beglowac You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
